### PR TITLE
Add virtual destructor to engine configs

### PR DESCRIPTION
**Pull request by dneto0**
Partially address #277

### CODE REVIEW

File: include/amber/amber_dawn.h [hunk 1]
```diff
@@ -22,6 +22,8 @@ namespace amber {
 
 /// Configuration for the Dawn engine.
 struct DawnEngineConfig : public EngineConfig {
+  virtual ~DawnEngineConfig() = default;
```
Comment (dj2) [inline include/amber/amber_dawn.h]: ~DawnEngineConfig() override = default;

File: include/amber/amber_vulkan.h [hunk 2]
```diff
@@ -27,6 +27,8 @@ namespace amber {
 /// Configuration for the Vulkan Engine. The following are all required when
 /// when using the vulkan backend.
 struct VulkanEngineConfig : public EngineConfig {
+  virtual ~VulkanEngineConfig() = default;
```
Comment (dj2) [inline include/amber/amber_vulkan.h]: ditto

File: include/amber/amber_dawn.h (hunk 1 above)
Comment (dneto0) [reply include/amber/amber_dawn.h]: done.

**Comment by dneto0** [comment]
Rebased.  Also put the destructor definition for each of the three classes into a .cc  in the right lib.  That should fix the clang complaints about not having a home for the vtable.

**Comment by dj2** [comment]
I'm going to close this so it drops out of the PR queue since it's not being worked on at the moment.

**Pull request closed without merging**